dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift

```diff
@@ -115,55 +132,70 @@ private class dydxTakeProfitStopLossViewPresenter: HostedViewPresenter<dydxTakeP
         viewModel?.oraclePrice = market?.oraclePrice?.doubleValue
     }
 
-    private func updateValidationErrors(_ errors: [ValidationError]) {
-        // TODO: update
-        if errors.isEmpty {
-            print("mmm: no errors")
-            viewModel?.takeProfitStopLossInputAreaViewModel?.takeProfitAlert = nil
-            viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossAlert = nil
-        } else {
-            for error in errors {
-                switch error.code {
-                case "TRIGGER_MUST_BELOW_INDEX_PRICE":
-                    print("mmm: TRIGGER_MUST_BELOW_INDEX_PRICE")
-                default:
-                    print("mmm: ", error.code)
+    private func update(triggerOrdersInput: TriggerOrdersInput?, errors: [ValidationError]) {
+
+        viewModel?.takeProfitStopLossInputAreaViewModel?.takeProfitAlert = nil
+        viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossAlert = nil
+
+        for error in errors {
+            let alert = InlineAlertViewModel(.init(title: error.resources.title?.localized, body: error.resources.text?.localized, level: .error))
+            switch error.code {
+            case "TRIGGER_MUST_ABOVE_INDEX_PRICE":
+                print("mmm: \(error)")
```

Comment:
FYI, the "error.fields" should now contains the TriggerOrdersInputField where the error occurs..  It's probably easier to check for that.

Reply:
👍  i will do that in a follow up PR